src/core/machine-id-setup.c

```diff
@@ -161,7 +166,9 @@ int machine_id_setup(const char *root, sd_id128_t machine_id, sd_id128_t *ret) {
 
         /* And now, let's mount it over */
         if (mount(run_machine_id, etc_machine_id, NULL, MS_BIND, NULL) < 0) {
-                (void) unlink_noerrno(run_machine_id);
+                /* We failed to mount over /etc/machine-id, but let's leave /run/machine-id
+                 * so we can commit it later if /etc/machine-id becomes writable
+                 * (this will be done by systemd-machine-id-commit.service) */
                 return log_error_errno(errno, "Failed to mount %s: %m", etc_machine_id);
         }
```

Comment:
Drop {}.

Reply:
got tricked by the multi-line comment, will do.